lib/big_ml/batch_centroid.rb

```diff
@@ -0,0 +1,39 @@
+require 'big_ml/base'
+
+module BigML
+  class BatchPrediction < Base
+    BATCH_PREDICTION_PROPERTIES = [
+      :category, :code, :created, :credits, :dataset, :dataset_status,
+      :description, :fields, :dataset, :model, :model_status, :name,
+      :objective_fields, :prediction, :prediction_path, :private, :resource,
+      :source, :source_status, :status, :tags, :updated
+    ]
+
+    attr_reader *BATCH_PREDICTION_PROPERTIES
+
+    class << self
+      def create(model_or_ensemble, dataset, options = {})
+        arguments = { dataset: dataset }
+        if model_or_ensemble.start_with? 'model'
+          arguments[:model] = model_or_ensemble
+        elsif model_or_ensemble.start_with? 'ensemble'
+          arguments[:ensemble] = model_or_ensemble
+        else
+          raise ArgumentError, "Expected model or ensemble, got #{model_or_ensemble}"
+        end
```

Comment:
I'll extract this code to its own class, f.e `AttributesBuillder` so all this logic will be encapsulated there.